### PR TITLE
feat(curriculum): add recursion quiz with 20 questions

### LINKED ISSUE #56214: [HACKTOBERFEST 2024] - Recursion Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Recursion quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-recursion`.

**All quiz questions need to relate to one of these topics:**

- Recursion
- base cases
- call stack
- tail-call and head-call

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by PRAteek-singHWY**
…actors

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56214

This pull request introduces a new quiz focused on recursion concepts, including base cases, call stacks, and types of recursion. The quiz consists of 20 questions designed to assess understanding and reinforce learning in recursive programming through multiple-choice and code block formats.

Key Changes:
Added 20 questions related to various recursion topics.
Included multiple short code block questions to enhance engagement and interactivity.
Ensured all questions have clear distractors and correct answers for effective assessment.
This quiz aims to provide learners with a comprehensive evaluation of their recursion knowledge, helping them solidify their understanding of recursive techniques.

### CODE REVIEW

**Comment by jeremylt** [comment]
Your pull request does not follow the contribution guidelines or how to open a pull request guide. Please read and follow those instructions.

**Comment by PRAteek-singHWY** [comment]
@naomi-lgbt can u please update me if there's any update regarding my PR and do i need to make something right .
Please inform me .

**Comment by ilenia-magoni** [comment]
@PRAteek-singHWY Naomi has not had the time to review your PR, it will be reviewed be assured, meanwhile please avoid pinging the maintainers

File: curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md [hunk 1]
```diff
@@ -17,439 +17,507 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder question
+What will be the output of the following function call: `fibonacci(5)`?
+
+```cpp
+int fibonacci(int n) {
+    if (n <= 1) return n;
+    return fibonacci(n - 1) + fibonacci(n - 2);
+}
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md:49]: please write your code in JavaScript

File: curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md [hunk 2]
```diff
@@ -17,439 +17,507 @@ Answer all of the questions below correctly to pass the quiz.
...
+
+
+```cpp
+void printNumbers(int n) {
+    if (n > 0) {
+        printNumbers(n - 1);
+        cout << n << " ";
+    }
+}
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md:110]:
````suggestion
In the following code, what does the recursive call represent?

```cpp
void printNumbers(int n) {
    if (n > 0) {
        printNumbers(n - 1);
        cout << n << " ";
    }
}
```
````
please also use JavaScript

File: curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md [hunk 3]
```diff
@@ -17,439 +17,507 @@ Answer all of the questions below correctly to pass the quiz.
...
+The program will run indefinitely.
 
 #### --answer--
 
-Placeholder answer
+The program will run indefinitely.
 
+### --question--
+
+#### --text--
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md:523]: what are these headers down here on their own? remove them

File: curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md [hunk 4]
```diff
@@ -17,439 +17,507 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --text--
 
-Placeholder question
+What is recursion in programming?
 
 #### --distractors--
 
-Placeholder distractor 1
+A programming technique where a function calls itself.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md:24]: You can't have a distractor be the same as the answer, change this

**Comment by ilenia-magoni** [comment]
I have left a few comments, you can use those to improve all your questions, thank you for your work

File: curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md (hunk 2 above)
Comment (PRAteek-singHWY) [reply curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md:110]: Respected maintainers ,All the changes have been made as requested
Please review it and tell me if there's any issue further .

File: curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md (hunk 4 above)
Comment (naomi-lgbt) [reply curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md:24]: This comment should not have been resolved. The issue is still present.

File: curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md (hunk 1 above)
Comment (naomi-lgbt) [reply curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md:49]: This comment should not have been resolved. The issue is still present.

File: curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md (hunk 2 above)
Comment (naomi-lgbt) [reply curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md:110]: This comment has not been resolved.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Please complete Ilenia's review suggestions.

**Comment by gikf** [comment]
Hey @PRAteek-singHWY,

did you have a chance to take a look at the feedback above?

**Pull request closed without merging**